__tests__/app.test.js

```diff
@@ -1,87 +1,199 @@
+        .expect(200)
+        .then(({ body }) => {
+          const { allTopics } = body;
+          allTopics.forEach((topic) => {
+            expect(topic.hasOwnProperty("description")).toBe(true);
+            expect(topic.hasOwnProperty("slug")).toBe(true);
+          });
+        });
+    });
+  });
+  describe("STATUS: 404", () => {
     test(" responds with a status code 404 if endpoint is incorrect/spelling erros", () => {
-        return request(app)
-      .get("/api/topic")
-      .expect(404)
-      .then(({ body }) => {
-        expect(body.msg).toBe("Not Found! Please check path.");
-      });
-    })
-   })
+      return request(app)
+        .get("/api/topic")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Not Found!");
+        });
+    });
+  });
 });
 
 describe("GET /api/articles/:article_id", () => {
-    describe("STATUS 200", () => {
-        test("responds with status 200 and returns relevant article", () => {
-            return request(app)
-            .get('/api/articles/1')
-            .expect(200)
-            .then((res) => {
-                const date = new Date(1594329060000).toJSON();
-              expect(res.body.article).toEqual({
-                username: 'butter_bridge',
-                article_id: 1,
-                title: "Living in the shadow of a great man",
-                topic: "mitch",
-                body: "I find this existence challenging",
-                created_at: date,
-                votes: 100
-              });
-            });
-        })
-    })
-    describe("STATUS 400s", () => {
-        test("Status 400 - invalid id", () => {
-            return request(app)
-            .get('/api/articles/invalidId')
-            .expect(400)
-            .then((res) => {
-            expect(res.body.msg).toBe('Invalid id');
-            });
-        })
-        test("STATUS 404 - article doesn't exist", () => {
-            return request(app).get('/api/articles/999').expect(404).then((res) => {
-            expect(res.body.msg).toBe("Article doesn't exist");
-             });
-        })
-    })
-})
\ No newline at end of file
+  describe("STATUS 200", () => {
+    test("responds with status 200 and returns relevant article", () => {
+      return request(app)
+        .get("/api/articles/1")
+        .expect(200)
+        .then((res) => {
+          const date = new Date(1594329060000).toJSON();
+          expect(res.body.article).toEqual({
+            username: "butter_bridge",
+            article_id: 1,
+            title: "Living in the shadow of a great man",
+            topic: "mitch",
+            body: "I find this existence challenging",
+            created_at: date,
+            votes: 100,
+          });
+        });
+    });
+  });
+  describe("STATUS 400s", () => {
+    test("Status 400 - invalid id", () => {
+      return request(app)
+        .get("/api/articles/invalidId")
+        .expect(400)
+        .then((res) => {
```

Comment:
if you're destructuring body from response in earlier tests, try to be consistent with your decision to do so